grid2op/Reward/_alertTrustScore.py

```diff
@@ -0,0 +1,168 @@
+        self.cumulated_reward = 0
+        #KPIs
+        self.total_nb_attacks = 0
+        self.nb_last_attacks=0
+
+        # TODO
+        #self.total_nb_alerts = 0
+        #self.alert_attack_no_blackout = 0
+        #self.alert_attack_blackout = 0
+        #self.no_alert_attack_no_blackout = 0
+        #self.no_alert_attack_blackout = 0
+        #self.blackout_encountered = False
+            
+    def __call__(self, action, env, has_error, is_done, is_illegal, is_ambiguous):
+
+        self.blackout_encountered = self.is_in_blackout(has_error, is_done)
+        
+        score_ep = 0.
+        if self._is_simul_env:
+            return score_ep
+            
+        res = super().__call__(action, env, has_error, is_done, is_illegal, is_ambiguous)
+        self.cumulated_reward += res
+
+        lines_attacked = env._time_since_last_attack == 0
+        self.total_nb_attacks += np.sum(lines_attacked)
+
+        # TODO
+        #lines_alerted_beforeattack = np.equal(env._time_since_last_alert, env._time_since_last_attack + 1) and lines_attacked
+        #self.total_nb_alerts += np.sum(lines_alerted_beforeattack)
+            
+        if not is_done:
+
+            # TODO
+            #lines_attacked_no_blackout = env._time_since_last_attack == SURVIVOR_TIMESTEPS
+#
+            #self.alert_attack_no_blackout += np.sum(lines_alerted_beforeattack[lines_attacked_no_blackout])
+            #self.no_alert_attack_no_blackout += np.sum(~lines_alerted_beforeattack[lines_attacked_no_blackout])
+            
+            return score_ep
+            
+        else:
+            self.nb_last_attacks=np.sum(self._ts_attack)
+            cm_reward_min_ep, cm_reward_max_ep = self._compute_min_max_reward(self.total_nb_attacks,self.nb_last_attacks)
+            score_ep = self._normalisation_fun(self.cumulated_reward, cm_reward_min_ep, cm_reward_max_ep,self.min_score,self.max_score)
+
+            # TODO
+            #if self.blackout_encountered:
+            #    lines_attacked_dangerzone = (env._time_since_last_attack >= 0) * (env._time_since_last_attack < SURVIVOR_TIMESTEPS)
+            #
+            #    self.alert_attack_blackout += 1. * any(lines_alerted_beforeattack[lines_attacked_dangerzone])
+            #    self.no_alert_attack_blackout += 1. * any(~lines_alerted_beforeattack[lines_attacked_dangerzone])
+            #else :
+            #    lines_attacked_no_blackout = env._time_since_last_attack > 0
+            #
+            #    self.alert_attack_no_blackout += np.sum(lines_alerted_beforeattack[lines_attacked_no_blackout])
+            #    self.no_alert_attack_no_blackout += np.sum(~lines_alerted_beforeattack[lines_attacked_no_blackout])
+            
+            return score_ep
+        
+    @staticmethod
+    def _normalisation_fun(cm_reward, cm_reward_min_ep, cm_reward_max_ep,min_score,max_score):
+        standardized_score = np.round((cm_reward - cm_reward_min_ep) / (cm_reward_max_ep - cm_reward_min_ep +1e-5),4)
```

Comment:
Why rounding ? It should not be rounded at this stage. If a user wants to round it it's his responsibility, not the one of the class.

Reply:
Because then you get score of 0.99993.
Where should we round this then ? Here it is to compensate for the addition of the tolerance that is artificial